src/tui/editor.zig

```diff
@@ -3924,6 +3925,49 @@ pub const Editor = struct {
     }
     pub const to_lower_meta = .{ .description = "Convert selection or word to lower case" };
 
+    fn switch_case_cursel(self: *Self, root_: Buffer.Root, cursel: *CurSel, allocator: Allocator) error{Stop}!Buffer.Root {
+        var root = root_;
+        const saved = cursel.*;
+        const sel = if (cursel.selection) |*sel| sel else ret: {
+            var sel = cursel.enable_selection();
+            move_cursor_right(root, &sel.begin, self.metrics) catch return error.Stop;
+            break :ret sel;
+        };
+        var sfa = std.heap.stackFallback(4096, self.allocator);
+        const cut_text = copy_selection(root, sel.*, sfa.get(), self.metrics) catch return error.Stop;
+        defer allocator.free(cut_text);
+        const cd = CaseData.init(allocator) catch return error.Stop;
+        defer cd.deinit();
+        const flipped = blk: {
+            var bytes = std.ArrayList(u8).initCapacity(allocator, cut_text.len) catch return error.Stop;
+            defer bytes.deinit();
+
+            var iter = code_point.Iterator{ .bytes = cut_text };
+            var buf: [4]u8 = undefined;
+
+            while (iter.next()) |cp| {
+                const code = if (cd.isLower(cp.code)) cd.toUpper(cp.code) else cd.toLower(cp.code);
+                const len = std.unicode.utf8Encode(code, &buf) catch return error.Stop;
+                bytes.appendSliceAssumeCapacity(buf[0..len]);
```

Comment:
This seems likely to panic. Who guarantees that the converted codepoints encode to exactly the same total byte length as the original string?